src/analyses/base.ml

```diff
@@ -2723,7 +2747,7 @@ struct
     | Alloca size, _ -> begin
         match lv with
         | Some lv ->
-          let heap_var = AD.of_var (heap_var true man) in
+          let heap_var = alloc Queries.AllocationLocation.Stack size in
```

Comment:
Since `alloca` isn't standard, it's behavior isn't too precisely described.
For example, whether it can fail for non-zero. The man page only says stack overflow is impossible to detect, so will it always appear to succeed?
Also the man page says nothing about zero argument. Seems like it could be a no-op and not bump the stack pointer at all. Surprisingly, a quick GCC Godbolt check reveals that `alloca(0)` compiles to a bit more than a no-op, but much less than `alloca(1)`.

The previous behavior here was to always succeed, so there will be a difference but I'm not sure how much it will matter. Maybe some sv-benchmarks use `alloca`.

Anyway, I think I'm fine with sharing the options for now, but this is something to keep in mind and maybe add a comment about.

Reply:
And in some cases, the null pointer is actually returned.

https://github.com/gcc-mirror/gcc/blob/a10723efcd611977c5e4a875b9cb1de920732d42/libiberty/alloca.c#L194-L196